cpg_infra/abstraction/azure.py

```diff
@@ -2,48 +2,187 @@
+        #         for threshold in self.config.budget_notification_thresholds
+        #     ]
+        #     kwargs['all_updates_rule'] = gcp.billing.BudgetAllUpdatesRuleArgs(
+        #         pubsub_topic=self.config.gcp.budget_notification_pubsub,
+        #         schema_version='1.0',
+        #     )
+
+        filters = budget_filter.pop('filter')
+        kwargs = dict(kwargs, dict(budget_filter))
+        
+        az.consumption.Budget(
+            resource_key,
+            budget_name=f'{project.name}-budget',
+            amount=budget,
+            category="Cost",
+            scope=f"/subscriptions/{self.subscription}/",
+            budget_filter=filters,
+            **kwargs,
+        )
+
+    def create_fixed_budget(
+        self,
+        resource_key: str,
+        *,
+        project,
+        budget: int,
+        start_date: date = date(2022, 1, 1),
+    ):
+        filters=az.consumption.BudgetFilterArgs(
+            and_=[
+                az.consumption.BudgetFilterPropertiesArgs(
+                    dimensions=az.consumption.BudgetComparisonExpressionArgs(
+                        name="ResourceId",
+                        operator="In",
+                        values=[project.id],
+                    ),
+                )
+            ]
+        )
+        return self.create_budget(
+            resource_key=resource_key,
+            project=project,
+            budget=budget,
+            budget_filter=az.consumption.BudgetArgs(
+                time_grain= 'Annually',
+                time_period= az.consumption.BudgetTimePeriodArgs(
+                    start_date=str(start_date),
+                    end_date= AZURE_BILLING_EXPIRY_DATE
+                ),
+                filter=filters
+            )
+        )
+
+    def create_monthly_budget(self, resource_key: str, *, project, budget: int):
+        # No start date here thats an issue
+        filters=az.consumption.BudgetFilterArgs(
+            and_=[
+                az.consumption.BudgetFilterPropertiesArgs(
+                    dimensions=az.consumption.BudgetComparisonExpressionArgs(
+                        name="ResourceId",
+                        operator="In",
+                        values=[project.id],
+                    ),
+                )
+            ]
+        )
+        return self.create_budget(
+            resource_key=resource_key,
+            project=project,
+            budget=budget,
+            budget_filter=az.consumption.BudgetArgs(
+                time_grain='Monthly',
+                time_period=az.consumption.BudgetTimePeriodArgs(
+                    start_date=AZURE_BILLING_START_DATE,
+                    end_date=AZURE_BILLING_EXPIRY_DATE
+                ),
+                filter=filters
+            )
         )
 
     def bucket_rule_undelete(self, days=UNDELETE_PERIOD_IN_DAYS) -> Any:
-        pass
+        az.storage.BlobServiceProperties(
```

Comment:
Return this?

Reply:
This needs a comment on why it's not being returned - or the method needs to be renamed? What happens if you call it twice with different days of the week - is it referenced specifically later?